### ISSUE TITLE
Kipf and Welling GCN paper

### ISSUE THREAD

**Issue by chippen001** [chippen001/Capstone-Resources#13]
### Title
Semi-Supervised Classification with Graph Convolutional Networks

### URL
https://arxiv.org/pdf/1609.02907.pdf 

### Summary 
This paper discusses an approach for semi-supervised learning on graph data, a convolutional neural network to operate directly on a graph. First-order approximation of spectral graph convolutions are used as the baseline of the convolutions. The model derivation is described in detail. The resulting model complexity is linear in the number of edges on the graph. The GCN described in this paper outperformed other baseline models in both accuracy and efficiency. 

### Key Points 
-Consider a multi-layer GCN with layer wise propagation rule motivated by a first-order approximation of localized spectral filters on graphs: H^(l+1) = σ(D^(-1/2)AD^(-1/2)H^(l)W^(l))

- A: Adjacency matrix of graph with self connections
- D: degree matrix (number of neighboring nodes, including self)
- W: layer-specific trainable weight matrix
- σ: activation function (ReLU)
- H: matrix of activations in layer

-Spectral convolutions on graphs: gθ ⋆x=UgθU^(T)x, however this is computationally expensive (O(N^2))
- Approximate in terms of Chebyshev polynomials.
- This K-localized convolution is now O(|E|) or linear in the number of edges. 
- Stack multiple of these convolutional layers to built neural network model for graph.
- Further approximate, constraining the number of parameters to address overfitting and minimize the amount of operations per layer. 
- Renormalization trick I+D^(−1/2)AD^(−1/2) →D ̃^(−1/2)A ̃D ̃^(−1/2) to prevent exploding or vanishing gradients.
- Z = D ̃^(−1/2) A ̃D ̃^(−1/2) XΘ

-Complexity is linear in the number of graph edges. 
-GCN performed better in accuracy efficiency (time) than other baseline methods compared. 
-Using the renormalization trick described in the paper for propagation model achieved best results.

### Citation
Thomas N. Kipf and Max Welling. Semi-supervised classification with graph convolutional networks. International Conference on Learning Representations, 2017.
